tempesta_fw/http_parser.c

```diff
@@ -140,22 +216,25 @@ do {									\
 done:									\
 	parser->state = __fsm_const_state;				\
 	/* Remaining number of bytes to process in the data chunk. */	\
-	parser->to_go = len - (size_t)(p - data);
+	parser->to_go = __data_remain(p);
 
-#define __FSM_MOVE_nf(to, n, field)				\
+#define __FSM_MOVE_nff(to, n, field, fixup)				\
 do {									\
 	p += n;								\
-	if (unlikely(p >= data + len || !*p)) {				\
+	if (unlikely(__data_offset(p) >= len)) {			\
 		r = TFW_POSTPONE; /* postpone to more data available */	\
 		__fsm_const_state = to; /* start from state @to next time */\
 		/* Close currently parsed field chunk. */		\
-		tfw_http_msg_field_chunk_fixup(msg, field, data, len);	\
+		if (fixup) __msg_field_fixup(field, data + len);	\
```

Comment:
I think coding style requires to place IF statement and its body on different lines.